lib/page.js

```diff
@@ -1,56 +1,61 @@
-const { every } = require('lodash');
+const { every, merge } = require('lodash');
+const { combineReducers } = require('redux');
 const path = require('path');
 const express = require('express');
 const match = require('minimatch');
-const persistQuery = require('./middleware/persist-query');
-const actions = require('./actions');
+const rootReducer = require('./reducers');
+
+const defaultMiddleware = (req, res, next) => next();
 
 module.exports = ({
-  name,
-  reducers,
   root,
-  schema,
-  pageContent
+  content = {},
+  locals = defaultMiddleware
 }) => {
   const app = express.Router();
 
-  const allowedSubPaths = ['/', '/assets/**', '/edit', '/edit/*', '/delete'];
-
-  app.use((req, res, next) => {
+  const _checkSubpath = (req, res, next) => {
+    const allowedSubPaths = ['/', '/assets/**', '/edit', '/edit/*', '/delete'];
     if (every(allowedSubPaths, p => !match(req.path, p))) {
       return next('router');
     }
     next();
-  });
+  };
 
   app.use('/assets', express.static(path.resolve(root, './dist')));
 
-  app.use((req, res, next) => {
-    res.locals.reducers = [ 'user', 'url', 'content', ...reducers ];
-    if (schema) {
-      res.store.dispatch(actions.setSchema(schema));
-    }
-    next();
-  });
-
-  app.use((req, res, next) => {
-    const filename = req.path.split('/').pop() || 'index';
+  const _locals = (req, res, next) => {
     const url = req.baseUrl === '/'
       ? ''
       : req.baseUrl;
-    res.store.dispatch(actions.setUrl(url));
-    res.locals.scripts = ['/public/bundle.js', `${url}/assets/${filename}/bundle.js`];
+    const filename = req.path.split('/').pop() || 'index';
+    const Component = require(path.resolve(root, `./views/${filename}`));
 
-    res.template = path.resolve(root, `views/${filename}`);
-    next();
-  });
+    let pageContent;
 
-  app.use((req, res, next) => {
-    res.store.dispatch(actions.setContent(pageContent));
-    next();
-  });
+    try {
+      pageContent = require(path.resolve(root, './content'));
+    } catch (err) {
+      pageContent = {};
+    }
 
-  app.use(persistQuery());
+    Object.assign(res.locals, {
+      scripts: ['/public/bundle.js', `${url}/assets/${filename}/bundle.js`],
+      Component: Component.default || Component,
+      rootReducer: combineReducers(rootReducer),
+      static: {
+        url,
+        content: merge({}, res.locals.static.content, pageContent, content)
+      }
+    });
+    res.template = path.resolve(__dirname, '../pages/common/views/index');
+    return locals(req, res, next);
```

Comment:
I don't think we need this insertion point since nothing comes after it. We can just let a page mount its own middleware once initialised.

Reply:
good point 👍